src/State.py

```diff
@@ -1,44 +1,34 @@
 
-from chess_enums import Space
-from chess_enums import Player
-from chess_enums import Piece
+from ChessEnums import Space
+from ChessEnums import Player
+from ChessEnums import Piece
 
 
 class State:
-    moves: list  # of Actions
+    moves: list = []  # of Actions
 
     def __init__(self) -> None:
-        board = [Piece.___]*len(Space)
-        board[Space.A2] = board[Space.B2] = board[Space.C2] = board[Space.D2] = board[
-            Space.E2] = board[Space.F2] = board[Space.G2] = board[Space.H2] = Piece.W_P
-        board[Space.B1] = board[Space.G1] = Piece.W_N
-        board[Space.C1] = board[Space.F1] = Piece.W_B
-        board[Space.A1] = board[Space.H1] = Piece.W_R
-        board[Space.D1] = Piece.W_Q
-        board[Space.E1] = Piece.W_K
+        self.board = [Piece.___]*len(Space)
+        self.board[Space.A2] = self.board[Space.B2] = self.board[Space.C2] = self.board[Space.D2] = self.board[
+            Space.E2] = self.board[Space.F2] = self.board[Space.G2] = self.board[Space.H2] = Piece.W_P
+        self.board[Space.B1] = self.board[Space.G1] = Piece.W_N
+        self.board[Space.C1] = self.board[Space.F1] = Piece.W_B
+        self.board[Space.A1] = self.board[Space.H1] = Piece.W_R
+        self.board[Space.D1] = Piece.W_Q
+        self.board[Space.E1] = Piece.W_K
 
-        board[Space.A7] = board[Space.B7] = board[Space.C7] = board[Space.D7] = board[
-            Space.E7] = board[Space.F7] = board[Space.G7] = board[Space.H7] = Piece.B_P
-        board[Space.B8] = board[Space.G8] = Piece.B_N
-        board[Space.C8] = board[Space.F8] = Piece.B_B
-        board[Space.A8] = board[Space.H8] = Piece.B_R
-        board[Space.D8] = Piece.B_Q
-        board[Space.E8] = Piece.B_K
+        self.board[Space.A7] = self.board[Space.B7] = self.board[Space.C7] = self.board[Space.D7] = self.board[
+            Space.E7] = self.board[Space.F7] = self.board[Space.G7] = self.board[Space.H7] = Piece.B_P
+        self.board[Space.B8] = self.board[Space.G8] = Piece.B_N
+        self.board[Space.C8] = self.board[Space.F8] = Piece.B_B
+        self.board[Space.A8] = self.board[Space.H8] = Piece.B_R
+        self.board[Space.D8] = Piece.B_Q
+        self.board[Space.E8] = Piece.B_K
 
-        self.king_space = {Space.E1, Space.E8}
-        self.king_moved = {False, False}
-        self.rookA_moved = {False, False}
-        self.rookH_moved = {False, False}
+        self.king_space = [ Space.E1, Space.E8 ]
+        self.king_moved = [ False, False ]
+        self.rookA_moved = [ False, False ]
+        self.rookH_moved = [ False, False ]
         self.player = Player(len(self.moves) % 2)
         self.check = False
-
-    def __eq__(self, par_state) -> bool:
```

Comment:
The eq function in python is actually a comparison operator, what was defined in the prototype was assignment. I believe class assignment in Python is not overloadable, so I removed this.